src/perform_step/dae_perform_step.jl

```diff
@@ -0,0 +1,36 @@
+function initialize!(integrator, cache::DImplicitEulerCache) end
+
+@muladd function perform_step!(integrator, cache::DImplicitEulerCache, repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack atmp,tmp = cache
+  alg = unwrap_alg(integrator, true)
+
+  # initial guess
+  guess = tmp
+  guess .= zero(eltype(u))
+
+  zero_func! = (out, x) -> begin
+                f(out, (x .- uprev) ./ dt, x, p, t)
+              end
+  u .= NLsolve.nlsolve(zero_func!, guess).zero
```

Comment:
The hard part will be trying to make this use our same nlsolver setup.

Reply:
yeah, some new overload of `nlsolve!` is required right now because the current setup expects explicit form very hard.